app/services/hyrax/workflow/sipity_actions_generator.rb

```diff
@@ -38,11 +39,39 @@ def call
       private
 
         def generate_state_diagram!
+          # look for old states that need to be removed and validate the states can be removed.
+          removed_states = validate_states_to_be_removed
```

Comment:
Is there some value in creating this variable at this position in the method? Or could we avoid the variable and change line 57 to `validate_states_to_be_removed.each(&:delete)`?